Sources/HaCWebsiteLib/Models/NewWorkshop+init.swift

```diff
@@ -0,0 +1,285 @@
+import Foundation
+import Yaml
+
+let filePaths = (
+  metadata: "/info/metadata.yaml",
+  description: "/info/description.md",
+  prerequisites: "/info/prerequisites.md",
+  promoImagesDirectory: "/info/promo_images",
+  notesDirectory: "/content/notes",
+  notesMarkdown: "/content/notes/notes.md",
+  examplesDirectory: "/examples",
+  presenterGuide: "/content/presenter_guide.md",
+  setupInstructions: "/info/setup_instructions.md"
+)
+
+let validImageExtensions = ["jpg", "svg", "png", "gif"]
+
+extension NewWorkshop {
+  enum Error: Swift.Error {
```

Comment:
Descriptive, fine-grained errors 😍 

Good thing Swift makes it so easy.

Reply:
Part of the motivation for these is the issue you mentioned of notifications. The initialiser here serves as a validator. When I was plugging I previous examples to the test cases I got nice errors that told me what the first error was. 
Keen for that to be utilised either through the website notifying or by making it easy to add this as a PR status check on workshop repos 👍